test/e2e/config_test.go

```diff
@@ -413,3 +413,89 @@ func asYAML(t *testing.T, val interface{}) string {
 	require.NoError(t, err)
 	return string(bs)
 }
+
+func TestDefaultConfig_PreserveExistingStatus(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	yaml := `
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.8.0
+  creationTimestamp: null
```

Comment:
Should we remove this and other such fields which are not relevant to the test (Just to make the tests smaller)?

Reply:
I can prune it back to the core essence. I included a full resource as generated by controller-gen because that's where I encountered the issue

Reply:
I think that having just the core essence should be good enough as the functionality effects only a specific section of the resource.